frontend/src/js/small-tab-navigation/HoverNavigatable.tsx

```diff
@@ -0,0 +1,41 @@
+import { FC } from "react";
+import { useDrop } from "react-dnd";
+import { DNDType } from "../common/constants/dndTypes";
+
+
+interface PropsT {
+    timeUntilNavigateOverride?: number;
+    triggerNavigate: () => void;
+    disabled?: boolean;
+}
+
+// default time until the hover triggers navigation
+// 600 ms is the default time for a tooltip to appear
+// feels very responsive
+// can be overriden by passing a timeUntilNavigateOverride prop
+const TIME_UNTIL_NAVIGATE = 600;
+
+export const HoverNagiatable:FC<PropsT> = ({ timeUntilNavigateOverride, triggerNavigate, children, disabled }) => {
```

Comment:
Going forward, let's try and avoid React.FC, see https://github.com/facebook/create-react-app/pull/8177. Preferrably, let's just inline props definitions like `const Comp = ({ a, b }: { a: string, b: number}) => {`